numpy/lib/format.py

```diff
@@ -242,6 +242,19 @@ def read_magic(fp):
         major, minor = magic_str[-2:]
     return major, minor
 
+class MetadataWarning(UserWarning):
+    pass
```

Comment:
See my updated comment - I'm not sure this is useful after all. If the file is corrupt and unreadable, the user has little motivation to suppress the warning.

Reply:
removing the specialized warning